Sources/Jobs/JobQueueHandler.swift

```diff
@@ -121,6 +121,9 @@ final class JobQueueHandler<Queue: JobQueueDriver>: Sendable {
                     count -= 1
                     logger.debug("Retrying Job")
                     self.updateJobMetrics(for: job.name, startTime: startTime, retrying: true)
+                    /// Should we enque the job with delay until instead of sleeping?
+                    /// This would give us the benefit of allowing jobs to be pushed now and get processed later
+                    try await Task.sleep(nanoseconds: self.backoff(attempts: count))
```

Comment:
Instead of sleeping, I think the followings should be done instead

```swift
self.queue.push(
    Job(
        id,
        parameters,
        delayUntil: Date(timeIntervalSince1970: self.backoff(attempts: count)),
        remainingAttempt: count
    )
)
```

Then after the decoding step above, we should guard

```swift
guard (delayUntil ?? .distantPast) < Date() else {
    logger.info("requeing job because it's not ready to process.")
    self.queue.push(
        Job(
            id,
            parameters,
            delayUntil: Date(timeIntervalSinceNow: self.backoff(attempts: count)),
            remainingAttempt: count
        )
    )
}
```

Reply:
Yeah if we do a retry with back off then yeah pushing the jobs back onto the queue would be preferable. But I haven't thought of a nice way to do delayed jobs in redis yet.